### ISSUE TITLE
Shows collection's spatial extent better (zoom / antimeridian)

### ISSUE THREAD

**Issue by christophfriedrich** [Open-EO/openeo-vue-components#18]
Zooming to the bbox of the rectangle isn't always a good idea (e.g. for the tiny collection of the R server somewhere in Southern Africa). But showing the entire world isn't always good either (e.g. a collection that covers the Netherlands would be very small on a world scale and thus hard to see).

We might want a smart feature that zooms according to the collection's covered area, e.g. to a Europe view for the Netherlands case.

**Comment by m-mohr** [comment]
I made some experiments. I think a reasonable and easy to implement compromise would be the following behavior:

Zoom to extent and then zoom out three levels, but only until level 3 is reached and never zoom closer than zoom level 8 or so.

That's still not the ideal solution for the small R back-end collections, but improves things by 5 or 6 zoom levels that a user would have to zoom out at least. In this case it would only be about 3 levels to zoom in/out for a good detail/overview map so that seems to be the best compromise.

**Comment by christophfriedrich** [comment]
Not having seen it in action I can't really tell, but this sounds reasonable (or _smarter_ like the title of this issue requests) because it ensures you always get a bit of context but can still see the area in question. As long as that also works for the edge case of global collections?

Regarding the tiny R backend collection: I don't think there's any good solution for this use case at all...

_OT: So many "zoom"s in your comment -- now I have an Ohrwurm :notes:_

**Comment by m-mohr** [comment]
For global collections it would zoom either to level 1 or 0, depending on the bbox. Of course, it doesn't zoom to -1 ;-)

> _OT: So many "zoom"s in your comment -- now I have an Ohrwurm 🎶_

Are you serious? NOW I also have one.... 🙄

**Comment by m-mohr** [comment]
I "fiddled" around a bit more. Solution:
Zoom level:
* <= 2: Keep as is
* <=5: zoom out once
* <= 8: zoom out twice
* else: zoom out three time, but at least until level 8

JS Fiddle with example bounding boxes from some openEO collections: https://jsfiddle.net/atec3dmk/3/

What do you think, @christophfriedrich ?

**Comment by christophfriedrich** [comment]
I like this generally, only the world-wide example should zoom out one level less IMO -- I don't like the doubling world and I don't think it's a problem (in this case) when the bbox fills almost the entire map.

i.e. after the `Math.min` part add `newZoom = Math.max(newZoom, 1);` (my suggestion)

**Comment by m-mohr** [comment]
This behavior depends on the map size. For 500px height zoom level 1 looks good, for less 300px not so much. As this is meant for general purpose, I guess it may be better to just set "noWrap" to true as you did on the Hub. Although, how does this work with bboxes that span the antimeridian?

**Comment by christophfriedrich** [comment]
Hm okay, then level 1 is "too risky" because these maps will (probably) rarely be more than 300px high, thus it should stay as it is.

> span the antimeridian

I just tested this with `const boundingBox = [150, -84, 150, 84];` -- doesn't work at all anyway :thinking: 

![grafik](https://user-images.githubusercontent.com/17781682/77419173-45ca6400-6dc8-11ea-8904-eb0921886931.png)

**Comment by m-mohr** [comment]
Yes, unfortunately. Leaflet requires numbers higher than 180/ lower than -180 for that an then it displays it one the duplicated part of the map. A solution would be to use something like https://github.com/briannaAndCo/Leaflet.Antimeridian to show correct bboxes over the antimeridian - unfortunately there's no npm package yet. I added a ToDo for this to the code and added nowrap to the map.

**Comment by christophfriedrich** [comment]
With noWrap off it can look a bit funny :thinking: 

![grafik](https://user-images.githubusercontent.com/17781682/77639844-ab018f00-6f59-11ea-90e4-288b571998be.png)

**Comment by m-mohr** [comment]
We won't solve all edge cases, but I guess for this case we could set nowrap to true if the zoom level by default is 0 or 1. Or we disable norwap at all. I don't see a big issue with "multiple" worlds shown...

**Comment by m-mohr** [comment]
Map background is now transparent by default, maybe that makes it look less strange?
Also, there's a callback onAfterMapInit that you can pass and use to customize the map the way you want.
